test/boost/s3_test.cc

```diff
@@ -205,3 +205,28 @@ SEASTAR_THREAD_TEST_CASE(test_client_readable_file) {
     f.close().get();
     cln->close().get();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_client_put_get_tagging) {
+    const sstring name(fmt::format("/{}/testobject-{}",
+                                   tests::getenv_safe("S3_PUBLIC_BUCKET_FOR_TEST"), ::getpid()));
+    auto client = s3::client::make(tests::getenv_safe("S3_SERVER_ADDRESS_FOR_TEST"), make_minio_config());
+    auto data = sstring("1234567890ABCDEF").release();
+    client->put_object(name, std::move(data)).get();
+    {
+        auto tagset = client->get_object_tagging(name).get0();
+        BOOST_CHECK(tagset.empty());
+    }
+    {
+        s3::client::TagSet expected_tagset{{"1", "one"}, {"2", "two"}};
+        client->put_object_tagging(name, expected_tagset).get();
+        auto actual_tagset = client->get_object_tagging(name).get0();
+        BOOST_CHECK(actual_tagset == expected_tagset);
+    }
+    {
+        client->delete_object_tagging(name).get();
+        auto tagset = client->get_object_tagging(name).get0();
+        BOOST_CHECK(tagset.empty());
+    }
+    client->delete_object(name).get();
```

Comment:
This test uses a non-unique name for the test object, and although it tries to delete the object at the end of the test, if there is an exception in the middle of the test we won't reach this deletion. I think we should try harder. An easy solution is to create a RAII object which invents a unique object name, and also deletes the S3 object when the RAII object is destroyed.

Reply:
please note, `BOOST_CHECK()` allows a test case to proceed even if one of the tests fails. but yeah, these methods under test can throw. i will add another commit to address your comment using RAII. as other tests in this source file have the same problem.